src/ripple/app/paths/impl/BookStep.cpp

```diff
@@ -68,6 +68,9 @@ class BookStep : public StepImp<TIn, TOut, BookStep<TIn, TOut>>
 
     boost::optional<Cache> cache_;
 
+    // Offers that were not used in the reverse pass because they becaue unfunded
```

Comment:
Yes, the bug only happens when the offer becomes unfunded in the same strand.

I found this bug due to the numeric instability bug in the deferred credits
table (also fixed in this PR). Ironically, that caused an offer to become
unfunded when _more_ funds were added to the offer's owner. I have not seen a
real payment trigger this bug.

The test case "testRmMadeUnfunded" in Offer.test.cpp is an example of how this
can happen. In general, it can happen when two steps in a strand output the same
issue and the same account is debited - either an offer owner or an account
step.